logging/test/mock_hub.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:sentry/sentry.dart';
+
+final fakeDsn = 'https://abc@def.ingest.sentry.io/1234567';
+
+class MockHub implements Hub {
+  final List<Breadcrumb> breadcrumbs = [];
+  final List<CapturedEvents> events = [];
+
+  @override
+  void addBreadcrumb(Breadcrumb crumb, {dynamic hint}) {
+    breadcrumbs.add(crumb);
+  }
+
+  @override
+  Future<SentryId> captureEvent(
+    SentryEvent event, {
+    dynamic stackTrace,
+    dynamic hint,
+    ScopeCallback? withScope,
+  }) async {
+    events.add(CapturedEvents(event, stackTrace));
+    return SentryId.newId();
+  }
+
+  // everything below is not needed
+
+  @override
+  void bindClient(SentryClient client) {}
+
+  @override
+  Future<SentryId> captureException(
+    dynamic throwable, {
+    dynamic stackTrace,
+    dynamic hint,
+    ScopeCallback? withScope,
+  }) {
+    // TODO: implement captureException
+    throw UnimplementedError();
```

Comment:
I'd rather return `return SentryId.empty()` tho, for all non implemented cases, the TODOs are not useful since we should not implement them anyway

Reply:
Sure 👍